rsp_jupyter_extensions/handlers/query.py

```diff
@@ -1,112 +1,288 @@
+            nb=fname.read_text()
+        else:
+            nb = self._get_tap_query_notebook(url)
+        await self.refresh_query_history()  # Opportunistic
+        return _write_notebook_response(nb, fname)
+
+    def _get_ts_query_notebook(
+        self,
+        org: str,
+        repo: str,
+        directory: str,
+        notebook: str,
+        params: dict[str,str]
+    ) -> str:
         """Ask times-square for a rendered notebook."""
+        # Since we know the path we don't have to crawl the base github
+        # response.
+        nb_url = f"github/{org}/{repo}/{directory}/{notebook}"
+
+        # Get the endpoint for the rendered URL
+        obj = self._ts_client.get(nb_url).json()
+        rendered_url = obj["rendered_url"]
+
+        # Retrieve that URL and return the textual response, which is the
+        # string representing the rendered notebook "in unicode", which
+        # means "a string represented in the default encoding".
+        return self._ts_client.get(rendered_url, params=params).text
 
-        # These are all constant for this kind of query
+
+    def _get_nublado_seeds_notebook(
+        self, notebook: str, params: dict[str,str]
+    ) -> str:
+        """Partially-curried function with invariant parameters filled in."""
         org = "lsst-sqre"
         repo = "nublado-seeds"
-        directory = "portal"
-        notebook = "query"
+        directory = "tap"
 
-        # Since we know the path we don't have to crawl the base github
-        # response
-        nb_url = f"github/{org}/{repo}/{directory}/{notebook}"
+        return self._get_ts_query_notebook(
+            org,
+            repo,
+            directory,
+            notebook,
+            params
+        )
 
-        # The only parameter we have is query_url, which is the Portal query
+    def _get_tap_query_notebook(self, url: str) -> str:
+        """Even-more-curried helper function for TAP query notebook."""
+        notebook = "query"
+        # The only parameter we have is query_url, which is the TAP query
         # URL
         params = {"query_url": url}
 
-        # Get the endpoint for the rendered URL
-        obj = self._client.get(nb_url).json()
-        rendered_url = obj["rendered_url"]
+        return self._get_nublado_seeds_notebook(
+            notebook,
+            params
+        )
 
-        # Retrieve that URL and return the textual response, which is the
-        # string representing the rendered notebook "in unicode", which I
-        # think means, "a string represented in the default encoding".
-        return self._client.get(rendered_url, params=params).text
+    def _get_query_all_notebook(self) -> str:
+        """Even-more-curried helper function for TAP history notebook."""
+        notebook = "history"
+        params: dict[str,str] = {}
+        return self._get_nublado_seeds_notebook(
+            notebook,
+            params
+        )
+
+
+
+    @tornado.web.authenticated
```

Comment:
Small lint issue with extra spaces

```suggestion

    @tornado.web.authenticated
```